### PR TITLE
Jh Add Course From Search

### LINKED ISSUE #7: FEATURE: Add sections to personal schedule directly from search
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

add a course/section from the search page to a personal schedule

### So that...

more easily add courses to my schedule

### Discussion

Currently, adding a course to a personal schedule requires remembering/copy pasting the enroll code. It would be much nicer for UX if you can create add courses from searches directly.

### Acceptance Criteria

- [x] There is a "add" or "save" button in the search results table.
- [x] Clicking this button opens a modal, to select an existing personal schedule to add this course to.

- [x] The schedule selection should be by _name_, not by id.
- [x] The "add" button only appears for sections, not courses (since currently only individual sections, not courses can be added).
- [x] *Bonus*: In the modal, there is a link to to the "add new personal schedule" page, so user can create one from there if they have not already.

### Implementation Todos

Frontend:
- [x] Create a react component that allows selecting from existing personal schedules by _name_, not by id. Note: the ID may be used internally in the code, just not visible to the user.
- [x] Create a modal/webpage for adding a new course. This should use the new schedule selection component.
- [x] In search results table, add an "add" button to applicable section that open the modal.

Backend:
- It _may_ be possible to implement the frontend without adding new backend endpoints. However, additional endpoints can be added if necessary.

Testing:
- [ ] Front end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by jonathan-herring**
In this PR, we add an add button for sections that allows users to add sections to one of their schedules. If they have no schedules, a link to a create schedule page is displayed.

image

Screen Shot 2024-03-10 at 11 43 09 PM

Closes #7

### CODE REVIEW

**Review by jiaqiguan2003 — DISMISSED** [review]
deployed the branch at: https://courses-jiaqiguan2003-dev.dokku-06.cs.ucsb.edu

all functions worked as expected! LGTM

**Review by Maria-Saucedo — DISMISSED** [review]
LGTM

File: frontend/package-lock.json
Comment (tallyhawley) [inline frontend/package-lock.json:1]: Don't push changes to package-lock.json

**Review by tallyhawley — CHANGES_REQUESTED** [review]
1. There is a bug where clicking "add" for the first time after logging in (or refreshing?) causes the "no schedule/ create a schedule" text to pop up even if I have schedules to choose from.
2. You should be able to add lectures with no sections -- e.g. ANTH 99. 
3. The schedule drop-down doesn't show the schedule quarter. (See the schedule drop-down on the "Add course" page)

**Review by Maria-Saucedo — APPROVED** [review]
Checked Deployment on and was able to properly add courses now. LGTM
https://courses-maria-saucedo-dev.dokku-06.cs.ucsb.edu/

**Review by tallyhawley — APPROVED** [review]
Looks great to me :)

**Review by aqiu28517 — COMMENTED** [review]
It's a bit odd that the save changes button shows even when you have no schedules to add to, but otherwise great. LGTM.

**Pull request merged**